src/com/company/SearchTheWord.java

```diff
@@ -0,0 +1,48 @@
+package com.company;
+
+import java.sql.SQLException;
+import java.util.StringTokenizer;
+
+import static com.company.ErrorMessage.SUCCESS_MESSAGE_WORD_FOUND;
+import static com.company.Main.*;
+
+public class SearchTheWord extends Thread {
+    String data;
+    int count;
+    String searchWord;
+    String txtFilePath;
+
+    public SearchTheWord(String data, int count, String searchWord, String txtFilePath) {
+        this.data = data;
+        this.count = count;
+        this.searchWord=searchWord;
+        this.txtFilePath=txtFilePath;
+    }
+
+    public void run() {
+        StringTokenizer st = new StringTokenizer(data);
+
+        while (st.hasMoreTokens()) {
+            if (searchWord.equalsIgnoreCase(st.nextToken())) {
+                count++;
+            }
+        }
+        if (count == 0) {
+            System.out.println(ErrorMessage.ERROR_MESSAGE_WORD_NOT_FOUND);
+            try {
+                database.databaseConnector(txtFilePath, searchWord, count, Constants.status, ErrorMessage.ERROR_MESSAGE_WORD_NOT_FOUND);
```

Comment:
Accessing the database field from Main is not the proper way to do it, you should accept database as an input parameter in the constructor and have a private field in SearchTheWord class itself

Reply:
Not addressed?

Reply:
done with this change.